### DISCUSSION TITLE
How we are going to fix energy management dashboard

### DISCUSSION THREAD

**Discussion by jobvk** [jobvk/Home-Assistant-Windcentrale#150] [General]
# Fix Energy Management Dashboard
The Windcentrale company has deactivated all live sensors, including one used to display data on the energy management dashboard. However, I am committed to maintaining this custom component and ensuring continued support for the energy management dashboard.

## The Problem
The issue at hand is that the Windcentrale company updates historical sensor data only a few hours after midnight from the previous day. While I can add a "day history sensor" to display data from the day before, and an "hourly history sensor," the resulting data update after midnight would show only one bar, which is not the desired outcome.

## Proposed Solution
My goal is to find a solution to update the statistics by the hour from the day before, ensuring accurate data on the energy dashboard. However, I have yet to discover a method for updating a sensor's statistics via a custom component. If you have any ideas to share, please do so.

**Comment by RonaldHass** [comment]
Would modeling the windmill's power based on wind speed (at its location) be accurate enough? That would -sort of- restore live data...

**Reply by jobvk** [reply]
The problem is the windspeed on the ground is a lot different then when is 40/65/80/85 meter in the air. You could calculate with this formula.

![image](https://github.com/jobvk/Home-Assistant-Windcentrale/assets/32730202/d8bb86cb-0633-4587-8228-f2b5d56228ea)

The reference wind speed v1 is measured at height h1. v2 is the wind speed at height h2. z0 is the roughness length (see table below).

Roughnessclass | Roughnesslength z0 | Land cover types
-- | -- | --
0 | 0.0002 m | Water surfaces: seas and Lakes
0.5 | 0.0024 m | Open terrain with smooth surface, e.g. concrete, airport runways, mown grass etc.
1 | 0.03 m | Open agricultural land without fences and hedges; maybe some far apart buildings and very gentle hills
1.5 | 0.055 m | Agricultural land with a few buildings and 8 m high hedges seperated by more than 1 km
2 | 0.1 m | Agricultural land with a few buildings and 8 m high hedges seperated by approx. 500 m
2.5 | 0.2 m | Agricultural land with many trees, bushes and plants, or 8 m high hedges seperated by approx. 250 m
3 | 0.4 m | Towns, villages, agricultural land with many or high hedges, forests and very rough and uneven terrain
3.5 | 0.6 m | Large towns with high buildings
4 | 1.6 m | Large cities with high buildings and skyscrapers

But the problem is then you have just the estimate windspeed and you have to calculate the energy and if you combine all the values I don't think it's going to be any accurate.

**Comment by gerard76** [comment]
I think we're better off voting in the next meeting. Hoping that there will be enough of us to make a change.

"Agendapunten voor ledenvergaderingen kan je melden in reactie op de uitnodiging voor een Algemene ledenvergadering."
